_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,366 @@
+---
+id: afw4f7
+name: Text node has minimum contrast
+rule_type: atomic
+
+description: |
+  This rule checks that the highest possible contrast of every text node with its background meets the minimal contrast requirement
+
+accessibility_requirements:
+  wcag20:1.4.3: # Contrast (Minimum)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
```

Comment:
This rule should also map to 1.4.6 (Contrast (enhanced)).
Essentially, failing 1.4.1 also fails 1.4.6. So if this rule fails, then 1.4.6 fails.

Reply:
I see your point, @Jym77 , though I think it is tidier to map this only to 1.4.3.

What do we usually do in these cases?

Reply:
I am not sure the case already arose in published rules. I can see at least two other PR that would be affected by a decision like that:
- #555: if the link (in context) is not descriptive, then the link (alone) is not descriptive. @carlosapaduarte chose to not have 2.4.9 as extra requirement.
- #1027: if the element has no parent with a language attribute, then the page itself has none. @patrickhlauke and me had some discussion about adding 3.1.1 as extra requirement: https://github.com/act-rules/act-rules.github.io/pull/1027#discussion_r349030552 and following

Both for this PR and for #555, it does make sense to write another rule for the AAA criterion afterwards. Even if AAA is not our main focus, these will essentially be "free" rules given that all the difficult stuff has already been discussed. For #1027, the "global" language rules already exist.

---
It seems that in each of these 3 cases, the rule writers decided to only map to the more restrictive criterion… So maybe it is indeed the way we should do 😄 
My idea was that the extra mapping is for free and does give more information to users and more "punch" to the rules…

Reply:
As per [ACT rule format](https://www.w3.org/TR/act-rules-format/#accessibility-requirements-mapping):
> the rule must list all accessibility requirements from those documents that are not satisfied when one or more of the outcomes of the rule is failed.

So I'd say that the format is actually requiring that we put both (here, adding 1.4.6).

Reply:
Hmmm… At the same time we have the habit that "passed examples must pass the SC", so if the rule also maps to 1.4.6 then all passed examples must have enhanced contrast… 🤔 

However, that habit is not part of the ACT rule format (as far as I can tell), while listing all requirements that fail when the rule fails is…